src/devices/Mcp3008/Mcp3008.cs

```diff
@@ -56,88 +60,102 @@ public void Dispose()
+                _controller.Write(_cs, PinValue.High);
+                _controller.Write(_clk, PinValue.Low);
+                _controller.Write(_cs, PinValue.Low);
 
                 for (var i = 0; i < 5; i++)
                 {
-                    if ((commandout & 0x80) > 0)
+                    if ((command & 0x80) > 0)
                     {
-                        _controller.Write(_MOSI, PinValue.High);
+                        _controller.Write(_mosi, PinValue.High);
                     }
                     else
                     {
-                        _controller.Write(_MOSI, PinValue.Low);
+                        _controller.Write(_mosi, PinValue.Low);
                     }
 
-                    commandout <<= 1;
-                    _controller.Write(_CLK, PinValue.High);
-                    _controller.Write(_CLK, PinValue.Low);
+                    command <<= 1;
+                    _controller.Write(_clk, PinValue.High);
+                    _controller.Write(_clk, PinValue.Low);
                 }
 
                 for (var i = 0; i < 12; i++)
                 {
-                    _controller.Write(_CLK, PinValue.High);
-                    _controller.Write(_CLK, PinValue.Low);
-                    trim_pot <<= 1;
+                    _controller.Write(_clk, PinValue.High);
+                    _controller.Write(_clk, PinValue.Low);
+                    result <<= 1;
 
-                    if (_controller.Read(_MISO) == PinValue.High)
+                    if (_controller.Read(_miso) == PinValue.High)
                     {
-                        trim_pot |= 0x1;
+                        result |= 0x1;
                     }
                 }
 
-                _controller.Write(_CS, PinValue.High);
+                _controller.Write(_cs, PinValue.High);
 
-                trim_pot >>= 1;
-                return trim_pot;
+                result >>= 1;
+                return result;
             }
         }
 
         // ported code from:
         // https://github.com/adafruit/Adafruit_Python_MCP3008/blob/master/Adafruit_MCP3008/MCP3008.py
-        private int ReadSpi(int adc_channel)
+        private int ReadSpi(int channel, InputConfiguration inputConfiguration)
         {
-            int command = 0b11 << 6;                  //Start bit, single channel read
-            command |= (adc_channel & 0x07) << 3;  // Channel number (in 3 bits)
-            var input = new Byte[] { (Byte)command, 0, 0 };
-            var output = new Byte[3];
+            byte configurationBits = GetConfigurationBits(channel, inputConfiguration);
+            byte[] input = new byte[] { configurationBits, 0, 0 };
+            byte[] output = new byte[3];
+
             _spiDevice.TransferFullDuplex(input, output);
 
-            var result = (output[0] & 0x01) << 9;
-            result |= (output[1] & 0xFF) << 1;
-            result |= (output[2] & 0x80) >> 7;
+            int result = (output[0] & 0b00000001) << 9;
```

Comment:
Any particular reason why you prefer to have this on binary instead of hex?

Reply:
This was my mistake and changed while reviewing bit moment (and counting fingers :smile:).  I overlooked this one and didn't revert.  I will change back to original this afternoon.